### PR TITLE
Improve consistency of hooks

### PR DESCRIPTION
**Pull request by basil**
Makes these hooks more consistent with each other.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/hook/JacocoHook.java [hunk 1]
```diff
@@ -18,7 +18,9 @@ public class JacocoHook extends PluginCompatTesterHookBeforeExecution {
     @Override
     public boolean check(@NonNull BeforeExecutionContext context) {
         Model model = context.getModel();
-        return "jacoco".equals(model.getArtifactId());
+        return "org.jenkins-ci.plugins".equals(model.getGroupId())
+                && "jacoco".equals(model.getArtifactId())
+                && "hpi".equals(model.getPackaging());
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/JacocoHook.java:23]: Elsewhere we check group ID, artifact ID, and packaging, so do so here as well.

File: src/main/java/org/jenkins/tools/test/hook/WarningsNGCheckoutHook.java [hunk 2]
```diff
@@ -22,8 +25,9 @@ protected String getParentFolder() {
     @Override
     public boolean check(@NonNull BeforeCheckoutContext context) {
         Model model = context.getModel();
-        return "warnings-ng-parent".equals(model.getArtifactId()) // localCheckoutDir
-                || "warnings-ng".equals(model.getArtifactId()); // checkout
+        return "io.jenkins.plugins".equals(model.getGroupId())
+                && ARTIFACT_IDS.contains(model.getArtifactId())
+                && "hpi".equals(model.getPackaging());
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/WarningsNGCheckoutHook.java:30]: Elsewhere we check group ID, artifact ID, and packaging, so do so here as well.

File: src/main/java/org/jenkins/tools/test/hook/WarningsNGExecutionHook.java [hunk 3]
```diff
@@ -10,10 +10,14 @@
...
+            Set.of(/* localCheckoutDir */ "warnings-ng-parent", /* checkout */ "warnings-ng");
+
     @Override
     public boolean check(@NonNull BeforeExecutionContext context) {
         Model model = context.getModel();
-        return "warnings-ng-parent".equals(model.getArtifactId()) // localCheckoutDir
-                || "warnings-ng".equals(model.getArtifactId()); // checkout
+        return "io.jenkins.plugins".equals(model.getGroupId())
+                && ARTIFACT_IDS.contains(model.getArtifactId())
+                && "hpi".equals(model.getPackaging());
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/WarningsNGExecutionHook.java:21]: Elsewhere we check group ID, artifact ID, and packaging, so do so here as well.

File: src/main/java/org/jenkins/tools/test/hook/WorkflowCpsHook.java [hunk 4]
```diff
@@ -23,17 +23,14 @@ protected String getPluginFolderName(@NonNull BeforeCheckoutContext context) {
...
-            // The history of groovy-cps history was merged into the repo, so the first multi-module
-            // release will be a little over 3500.
-            VersionNumber multiModuleSince = new VersionNumber("3500");
-            return "org.jenkins-ci.plugins.workflow".equals(model.getGroupId())
-                    && "workflow-cps".equals(model.getArtifactId())
-                    && pluginVersion.isNewerThan(multiModuleSince)
-                    && "hpi".equals(model.getPackaging());
-        }
-        return false;
+        VersionNumber pluginVersion = new VersionNumber(context.getPlugin().version);
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/WorkflowCpsHook.java:26]: This null check was redundant: we always set the plugin and version.

File: src/main/java/org/jenkins/tools/test/hook/JacocoHook.java (hunk 1 above)
Comment (jtnord) [reply src/main/java/org/jenkins/tools/test/hook/JacocoHook.java:23]: elsewhere we are going the wrong way - we should just check the `pluginID` as we are in the business of checking plugins not artifacts, but 🤷 
aka  `context.getPlugin().name.equals("jacoco");`

**Review by jtnord — APPROVED** [review]

File: src/main/java/org/jenkins/tools/test/hook/JacocoHook.java (hunk 1 above)
Comment (basil) [reply src/main/java/org/jenkins/tools/test/hook/JacocoHook.java:23]: I guess we just have different philosophies. I prefer to be as paranoid as possible, possibly from my days doing kernel development.

**Pull request merged**